### ISSUE TITLE
[Bug]: VM Disk Creation error

### ISSUE THREAD

**Issue by xriot2** [orange-cloudavenue/terraform-provider-cloudavenue#756]
### Terraform Core Version

v1.7.4

### Cloud Avenue Provider Version

0.18.4

### Affected Resource(s)

CloudAvenue Disk

### Expected Behavior

SCSI Disk creation (attached disk)

### Actual Behavior

error during the creation:
```
cloudavenue_vm.res_vm[0]: Still creating... [2m10s elapsed]
cloudavenue_vm.res_vm[0]: Creation complete after 2m13s [id=urn:vcloud:vm:a571b842-388d-49a1-9688-505ba1e0f515]
cloudavenue_vm_disk.disk[0]: Creating...
cloudavenue_vm_disk.disk[0]: Still creating... [10s elapsed]
cloudavenue_vm_disk.disk[0]: Still creating... [20s elapsed]
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to cloudavenue_vm_disk.disk[0], provider
│ "provider[\"registry.terraform.io/orange-cloudavenue/cloudavenue\"]"
│ produced an unexpected new value: .bus_type: was cty.StringVal("SCSI"), but
│ now cty.StringVal("SATA").
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
Uploading artifacts for failed job
```

### Terraform Configuration Files
```
resource "cloudavenue_vm_disk" "disk" {

  depends_on = [cloudavenue_vm.res_vm]

  count           = var.vm_config.count

  size_in_mb      = var.disk_config.disk_size_in_gb * 1024
  bus_type        = var.disk_config.bus_type
  bus_number      = var.disk_config.bus_number
  is_detachable   = var.disk_config.disk_is_detachable
  name            = var.disk_config.disk_is_detachable ? "${local.disk_name_prefix}${format("%02d", count.index)}" : null
  storage_profile = var.disk_config.storage_profile
  unit_number     = var.disk_config.unit_number
  vapp_id         = null
  vapp_name       = try(cloudavenue_vm.res_vm[count.index].vapp_name, null)
  vm_id           = null
  vm_name         = try(cloudavenue_vm.res_vm[count.index].name, null)
}

with variables

disk_config = {
    bus_type = "SCSI"
    bus_number = 0
    unit_number = 1
    disk_is_detachable = false
    storage_profile = "silver"
    disk_size_in_gb = 10
    disk_mount_point = "/var/lib/vital"
}
```

### Steps to Reproduce

just request a creation of vm with an attached disk with SCSI bus

### Would you like to implement a fix?

None

**Comment by azrod** [comment]
Hello

Please add terraform configuration for cloudavenue_vm resource 

Thanks

**Comment by xriot2** [comment]
Hi,
any news about this problem ?
best regards

**Comment by xriot2** [comment]
It's the same behaviour with the configuration:

disk_config = {
    bus_type = "SCSI"
    bus_number = 2
    unit_number = 0
    disk_is_detachable = false
    storage_profile = "silver"
    disk_size_in_gb = 10
    disk_mount_point = "/var/lib/vital"
}



cloudavenue_vm.res_vm[0]: Creation complete after 3m35s [id=urn:vcloud:vm:eb438227-a417-4584-b3c4-189b2287689b]
cloudavenue_vm_disk.disk[0]: Creating...
cloudavenue_vm_disk.disk[0]: Still creating... [10s elapsed]
cloudavenue_vm_disk.disk[0]: Still creating... [20s elapsed]
cloudavenue_vm_disk.disk[0]: Still creating... [30s elapsed]
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to cloudavenue_vm_disk.disk[0], provider
│ "provider[\"registry.terraform.io/orange-cloudavenue/cloudavenue\"]"
│ produced an unexpected new value: .bus_type: was cty.StringVal("SCSI"), but
│ now cty.StringVal("SATA").
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵

**Comment by xriot2** [comment]
Hi,
any news about this problem because we are blocked since a long time ?
best regards

**Comment by azrod** [comment]
Hi, 

Sorry for the delay in response. 

The problem does not seem to come from the terraform provider.

When you define "SCSI" what is the value actually defined on the IHM ?

Regards

**Comment by xriot2** [comment]
into the GUI, it indicates 'SATA' and not 'SCSI'

**Comment by vivekanandita** [comment]
I agree with @xriot2. I'm also facing same issue, when trying to create disk. 

earlier I used to use vcd provider disk schema was like below and it worked fine.
[bus_type](https://registry.terraform.io/providers/vmware/vcd/latest/docs/resources/vm_internal_disk#bus_type-1) - (Required) The type of disk controller. Possible values: ide, parallel( LSI Logic Parallel SCSI), sas(LSI Logic SAS (SCSI)), paravirtual(Paravirtual (SCSI)), sata, nvme. Note nvme requires

for CloudAvenue it is different as below:

[bus_type](https://registry.terraform.io/providers/orange-cloudavenue/cloudavenue/latest/docs/resources/vm_disk#bus_type-2) (String) (ForceNew) The type of disk controller. Attribute require replacement if is_detachable is false. Value must be one of : IDE, SATA, SCSI, NVME. Value defaults to SCSI.

Note: on my GUI portal for New cloud avenue I have bus_type values as below terraform provider on cloud avenue does not have any option to provide SCSI as bus_type:
values: 
**ide, 
parallel( LSI Logic Parallel SCSI), 
sas(LSI Logic SAS (SCSI)), 
paravirtual(Paravirtual (SCSI)), 
sata, 
nvme**

**Comment by azrod** [comment]
Hello, 

Sorry for the delay. 

Can your try to set NVME in your terraform ? 

Thanks

**Comment by azrod** [comment]
@xriot2 Many thanks for your test. 

I'm currently working to try to fix this problem.

**Comment by xriot2** [comment]
I've switch on version 0.26.0 for the source orange-cloudavenue/cloudavenue

I've tested in scsi and nvme.

It's now ok with nvme but for information the error is always the same for scsi bus_type setting:

cloudavenue_vm.res_vm[0]: Creation complete after 3m15s [id=urn:vcloud:vm:58c1d448-5ee1-49db-afd7-8b641d3ef81f]
cloudavenue_vm_disk.disk[0]: Creating...
cloudavenue_vm_disk.disk[0]: Still creating... [10s elapsed]
cloudavenue_vm_disk.disk[0]: Still creating... [20s elapsed]
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to cloudavenue_vm_disk.disk[0], provider
│ "provider[\"registry.terraform.io/orange-cloudavenue/cloudavenue\"]"
│ produced an unexpected new value: .bus_type: was cty.StringVal("SCSI"), but
│ now cty.StringVal("SATA").

The update is now sufficient for our ubuntu case.

**Comment by azrod** [comment]
Hello @xriot2,

In your test the disk `is_detachable` ?

**Comment by xriot2** [comment]
no

**Comment by azrod** [comment]
@xriot2 

Please try why release v0.26.1
